ndk/src/surface_texture.rs

```diff
@@ -0,0 +1,112 @@
+//! Bindings for [`ffi::ASurfaceTexture`]
+#[cfg(feature = "api-level-28")]
+use crate::native_window::NativeWindow;
+#[cfg(feature = "api-level-28")]
+use jni_sys::{jobject, JNIEnv};
+use std::ptr::NonNull;
+use thiserror::Error;
+
+#[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct SurfaceTexture {
+    ptr: NonNull<ffi::ASurfaceTexture>,
+}
+
+unsafe impl Send for SurfaceTexture {}
+
+#[derive(Debug, Error)]
+pub struct PosixError(pub i32);
+
+impl std::fmt::Display for PosixError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "Posix Error: {}", self.0)
+    }
+}
+
+impl Drop for SurfaceTexture {
+    fn drop(&mut self) {
+        unsafe { ffi::ASurfaceTexture_release(self.ptr.as_ptr()) }
+    }
+}
+
+impl SurfaceTexture {
+    /// Assumes ownership of `ptr`
+    ///
+    /// # Safety
+    /// `ptr` must be a valid pointer to an Android [`ffi::ASurfaceTexture`].
+    pub unsafe fn from_ptr(ptr: NonNull<ffi::ASurfaceTexture>) -> Self {
+        Self { ptr }
+    }
+
+    #[cfg(feature = "api-level-28")]
```

Comment:
Nit: typically write attributes _below_ the doc comment (but in this case I don't think we should have these at all, and just `cfg()` once on the entire module altogether).